### ISSUE TITLE
Watcher doesn't detect changes when artifacts are outside the directory from where skaffold is run

### ISSUE THREAD

**Issue by doubliez** [GoogleContainerTools/skaffold#2613]
### Expected behavior

When a file is changed in a workspace (directory specified in `build.artifacts[].context` option), a rebuild should be triggered automatically, even if the workspace is located outside the directory from where `skaffold dev` is run.

### Actual behavior

File change doesn't trigger a rebuild in this case. Note that this used to work but not sure in which version of Skaffold this got broken.

### Information

- Skaffold version: v0.35.0
- Operating system: macOS 10.14.6
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1beta13
kind: Config
build:
  artifacts:
  - image: myapplication
    context: ../../../../myapplication
deploy:
  kubectl:
    manifests:
    - ../../apps/myapplication.yaml
```

### Steps to reproduce the behavior

Run `skaffold dev` from the directory which has the `skaffold.yaml` file above. Note that the artifact context is located in a different directory which is not under the directory which contains `skaffold.yaml`.

Note that if I instead run skaffold from the `myapplication` directory, pointing to the `skaffold.yaml` using the `-f` option (and changing the paths accordingly in the config), file watching works fine.

However that's not an appropriate workaround as I have a lot of different artifacts that I want to be able to build and watch, while maintaining the `skaffold.yaml` config in a different directory.

And as I mentioned above, this used to work but got broken in some later version of Skaffold.

**Comment by JiaboHou** [comment]
I'm also able to reproduce this in my own setup. I believe I have been encountering this issue since v0.34.0.

**Comment by bjesus** [comment]
Also seeing this issue with Skaffold 0.35 on Arch Linux

**Comment by priyawadhwa** [comment]
Hey @doubliez thanks for submitting this bug -- I have a PR open to fix this issue (#2614).
